Mathlib/CategoryTheory/Localization/Triangulated.lean

```diff
@@ -0,0 +1,204 @@
+  clear a b fac hT₁ hT₂ T₁ T₂
+  intro T₁ T₂ hT₁ hT₂ a b fac
+  obtain ⟨α, hα⟩ := exists_leftFraction L W a
+  obtain ⟨β, hβ⟩ := (MorphismProperty.RightFraction.mk α.s α.hs T₂.mor₁).exists_leftFraction
+  obtain ⟨γ, hγ⟩ := exists_leftFraction L W (b ≫ L.map β.s)
+  have := inverts L W β.s β.hs
+  have := inverts L W γ.s γ.hs
+  dsimp at hβ
+  obtain ⟨Z₂, σ, hσ, fac⟩ := (MorphismProperty.map_eq_iff_postcomp L W
+    (α.f ≫ β.f ≫ γ.s) (T₁.mor₁ ≫ γ.f)).1 (by
+      rw [← cancel_mono (L.map β.s), assoc, assoc, hγ, ← cancel_mono (L.map γ.s),
+        assoc, assoc, assoc, hα, MorphismProperty.LeftFraction.map_comp_map_s,
+        ← Functor.map_comp] at fac
+      rw [fac, ← Functor.map_comp_assoc, hβ, Functor.map_comp, Functor.map_comp,
+        Functor.map_comp, assoc, MorphismProperty.LeftFraction.map_comp_map_s_assoc])
+  simp only [assoc] at fac
+  obtain ⟨Y₃, g, h, hT₃⟩ := Pretriangulated.distinguished_cocone_triangle (β.f ≫ γ.s ≫ σ)
+  let T₃ := Triangle.mk (β.f ≫ γ.s ≫ σ) g h
+  change T₃ ∈ distTriang C at hT₃
+  have hβγσ : W (β.s ≫ γ.s ≫ σ) := W.comp_mem _ _ β.hs (W.comp_mem _ _ γ.hs hσ)
+  obtain ⟨ψ₃, hψ₃, hψ₁, hψ₂⟩ := MorphismProperty.compatible_with_triangulation
+    T₂ T₃ hT₂ hT₃ α.s (β.s ≫ γ.s ≫ σ) α.hs hβγσ (by dsimp [T₃]; rw [reassoc_of% hβ])
+  let ψ : T₂ ⟶ T₃ := Triangle.homMk _ _ α.s (β.s ≫ γ.s ≫ σ) ψ₃
+    (by dsimp [T₃]; rw [reassoc_of% hβ]) hψ₁ hψ₂
+  have : IsIso (L.mapTriangle.map ψ) := Triangle.isIso_of_isIsos _
+    (inverts L W α.s α.hs) (inverts L W _ hβγσ) (inverts L W ψ₃ hψ₃)
+  refine' ⟨L.mapTriangle.map (completeDistinguishedTriangleMorphism T₁ T₃ hT₁ hT₃ α.f
+      (γ.f ≫ σ) fac.symm) ≫ inv (L.mapTriangle.map ψ), _, _⟩
+  · rw [← cancel_mono (L.mapTriangle.map ψ).hom₁, ← comp_hom₁, assoc, IsIso.inv_hom_id, comp_id]
+    dsimp [ψ]
+    rw [hα, MorphismProperty.LeftFraction.map_comp_map_s]
+  · rw [← cancel_mono (L.mapTriangle.map ψ).hom₂, ← comp_hom₂, assoc, IsIso.inv_hom_id, comp_id]
+    dsimp [ψ]
+    simp only [Functor.map_comp, reassoc_of% hγ,
+      MorphismProperty.LeftFraction.map_comp_map_s_assoc]
+
+/-- The pretriangulated structure on the localized category. -/
+def pretriangulated : Pretriangulated D where
+  distinguishedTriangles := L.essImageDistTriang
+  isomorphic_distinguished _ hT₁ _ e := L.essImageDistTriang_mem_of_iso e hT₁
+  contractible_distinguished :=
+    have := essSurj L W; L.contractible_mem_essImageDistTriang
+  distinguished_cocone_triangle f := distinguished_cocone_triangle L W f
+  rotate_distinguished_triangle := L.rotate_essImageDistTriang
+  complete_distinguished_triangle_morphism := complete_distinguished_triangle_morphism L W
+
+lemma isTriangulated_functor :
+    letI : Pretriangulated D := pretriangulated L W; L.IsTriangulated :=
+    letI : Pretriangulated D := pretriangulated L W; ⟨fun T hT => ⟨T, Iso.refl _, hT⟩⟩
+
+lemma essSurj_mapArrow : L.mapArrow.EssSurj  :=
+  essSurj_mapArrow_of_hasLeftCalculusofFractions L W
```

Comment:
Why is this restated?

Reply:
I found it convenient to restate it in the `Triangulated.Localization` namespace.

Reply:
In that case, maybe put the `alias` below the original lemma or just rename it altogether?

Reply:
I have renamed ` essSurj_mapArrow_of_hasLeftCalculusOfFractions` as `essSurj_mapArrow` (and kept `essSurj_mapArrow_of_hasRightCalculusofFractions`).